### PR TITLE
Add 3.1 release manifest

### PR DESCRIPTION
**Pull request by torcolvin**
Should I also add a file for 

manifest/3.1/3.1.0.xml with duplicate contents of 3.1.xml or leave that to the inevitable 3.1.1 point release?

I was going to merge this change into the release branch AND master.

### CODE REVIEW

**Comment by bbrks** [comment]
- `3.1.0.xml` comes only when we have a new `3.1.x` release (otherwise you'd end up with 2 `3.1.0` manifests)
- Nothing reads the branch version of `product-config.json` - `master` is the source of truth for what versions to build. Changing this isn't nessesary.
- We can get the `3.1.0.xml` into the release branch for our weird Jenkins jobs that depend on the release branch - but the build team's jobs read the `master` version.

**Comment by torcolvin** [comment]
> * Nothing reads the branch version of `product-config.json` - `master` is the source of truth for what versions to build. Changing this isn't nessesary.

Do you prefer to have release branch match master? Is it correct in theory that none of this needs to be done on the release branch?

File: manifest/3.1.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  <project name="product-texts" path="product-texts" remote="couchbase"/>
+  <project name="build" path="cbbuild" remote="couchbase" revision="844e2484067053d7af88e706a814766265bf03df">
+    <annotation name="VERSION" value="3.1.0"     keep="true"/>
+    <annotation name="BLD_NUM" value="@BLD_NUM@" keep="true"/>
+    <annotation name="RELEASE" value="@RELEASE@" keep="true"/>
+  </project>
+
+
+  <!-- Sync Gateway -->
+   <project name="sync_gateway" path="sync_gateway" remote="couchbase" />
```
Comment (bbrks) [inline manifest/3.1.xml]: We need to let them (the build team's job) know where to find 3.1.0 in SG's repo here:

```suggestion
   <project name="sync_gateway" path="sync_gateway" remote="couchbase" revision="release/3.1.0"/>
```

File: manifest/product-config.json
Comment (bbrks) [inline manifest/product-config.json:1]: The `manifest/default.xml` entry (starting line 4) needs updating to `"release": "3.2.0",` now

File: manifest/default.xml
Comment (bbrks) [inline manifest/default.xml:1]: Also need to make this change in `base/version.go` (`ProductAPIVersionMinor = "2"`)

File: manifest/product-config.json [hunk 2]
```diff
@@ -389,6 +389,15 @@
...
             "start_build": 1
         },
+        "manifest/3.1.xml": {
+            "release": "3.1.0",
+            "release_name": "Couchbase Sync Gateway 3.1.0",
+            "production": true,
+            "interval": 120,
+            "go_version": "1.19.5",
+            "trigger_blackduck": true,
+            "start_build": 1
```
Comment (bbrks) [inline manifest/product-config.json]: `"start_build"` needs to be `+1` of the current build numebr of 3.1.0 to ensure any subsequent builds get assigned the next build number.

At time of writing the latest build is `585`: 
http://latestbuilds.service.couchbase.com/builds/latestbuilds/sync_gateway/3.1.0/

**Review by bbrks — CHANGES_REQUESTED** [review]
A couple of things (these are in the doc)

File: manifest/3.1.xml (hunk 1 above)
Comment (bbrks) [reply manifest/3.1.xml]: this wasn't fixed

File: manifest/3.1.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+Use of this software is governed by the Business Source License included in
+the file licenses/BSL-Couchbase.txt.  As of the Change Date specified in that
+file, in accordance with the Business Source License, use of this software will
+be governed by the Apache License, Version 2.0, included in the file
+licenses/APL2.txt.
+-->
+
+<manifest>
+
+  <remote fetch="https://github.com/couchbase/" name="couchbase" revision="release/3.1.0"/>
```
Comment (bbrks) [inline manifest/3.1.xml]: remotes don't have revisions

```suggestion
  <remote fetch="https://github.com/couchbase/" name="couchbase"/>
```

**Review by bbrks — CHANGES_REQUESTED** [review]
the revision was added in the wrong place in the manifest

**Review by bbrks — APPROVED** [review]

**Pull request merged**